driver-core/src/main/com/mongodb/internal/TimeoutContext.java

```diff
@@ -166,29 +161,15 @@ private boolean hasTimedOutForCommandExecution() {
      * @return timeout to use.
      */
     public long timeoutOrAlternative(final long alternativeTimeoutMS) {
```

Comment:
This timeoutOrAlternative method should ideally be inlined, with `getMaxCommitTimeMS`, `getReadTimeoutMS`, and `getWriteTimeoutMS` encapsulating the result using lambdas, as was done in `runMaxTimeMSTimeout`. This requires more associated refactoring than seems appropriate for this PR.